### PR TITLE
Add experiment section on create pipeline run page

### PR DESCRIPTION
**Pull request by DaoDaoNoCode**
JIRA: https://issues.redhat.com/browse/RHOAIENG-2990

## Description

This PR does the following things:
1. Add a section named `Experiment` on the create run page to allow create a run with the selected experiment

Screenshot 2024-02-20 at 2 08 26 PM
Screenshot 2024-02-20 at 2 09 43 PM

2. Add the modal to allow create a new experiment

Screenshot 2024-02-20 at 2 09 02 PM

3. Allow duplicating the experiment information as well when duplicating a run

## How Has This Been Tested?

1. Go to the create run page
2. Test if the experiment select menu works
3. Try to create a new experiment
4. Try to create a run with the selected experiment
5. Try to duplicate the run to see if the experiment is pre-filled

## Test Impact

Update the cypress test to verify the experiment section works as expected.

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/concepts/pipelines/content/createRun/contentSections/ExperimentSection.tsx [hunk 1]
```diff
@@ -14,54 +17,40 @@ type ExperimentSectionProps = {
...
-        isOpen={openCreate}
-        onClose={(experiment) => {
-          if (experiment) {
-            refreshAllAPI();
-            onChange(experiment);
-          }
-          setOpenCreate(false);
-        }}
-      />
+      {openCreate && (
```
Comment (andrewballantyne) [inline frontend/src/concepts/pipelines/content/createRun/contentSections/ExperimentSection.tsx:43]: Avoid doing this -- it causes weird modal issues. Open/Close via a prop

File: frontend/src/concepts/pipelines/content/experiment/ManageExperimentModal.tsx [hunk 2]
```diff
@@ -15,11 +15,10 @@ import useCreateExperimentData from '~/concepts/pipelines/content/experiment/use
 import { ExperimentKFv2 } from '~/concepts/pipelines/kfTypes';
 
 type ManageExperimentModalProps = {
-  isOpen: boolean;
   onClose: (experiment?: ExperimentKFv2) => void;
 };
 
-const ManageExperimentModal: React.FC<ManageExperimentModalProps> = ({ isOpen, onClose }) => {
```
Comment (andrewballantyne) [inline frontend/src/concepts/pipelines/content/experiment/ManageExperimentModal.tsx:22]: 🤔 why did you change this?

File: frontend/src/concepts/pipelines/content/createRun/utils.ts [hunk 3]
```diff
@@ -39,6 +39,7 @@ const runTypeSafeDates = (runType: RunFormData['runType']): boolean =>
 
 export const isFilledRunFormData = (formData: RunFormData): formData is SafeRunFormData =>
   !!formData.nameDesc.name &&
+  !!formData.experiment &&
```
Comment (Gkrumbach07) [inline frontend/src/concepts/pipelines/content/createRun/utils.ts:42]: do you need experiment. wont it just default to `Default`. Do we want to pre-fill the field with default? Does have an opinion on this?

**Review by Gkrumbach07 — COMMENTED** [review]
works as intended. just the one comment

**Review by andrewballantyne — CHANGES_REQUESTED** [review]
Need to test this -- here is the code feedback.

**Comment by andrewballantyne** [comment]
Looking reasonable outside of that -- but I wonder if we don't want to default select `Default` Experiment 🤔 That could be something we swing back with.

File: frontend/src/__mocks__/mockKFExperiment.ts [hunk 4]
```diff
@@ -9,3 +9,58 @@ export const mockKFExperiment = (experiment?: Partial<ExperimentKFv2>): Experime
   storage_state: StorageStateKFv2.AVAILABLE,
   ...experiment,
 });
+
+export const mockExperimentsV2Proxy: { experiments: ExperimentKFv2[]; total_size: number } = {
```
[the remark this answers is not in the document]
Comment (DaoDaoNoCode) [reply frontend/src/__mocks__/mockKFExperiment.ts:13]: I think I was following a wrong pattern in the pipelines mock data, will clean that up. Thanks!

File: frontend/src/concepts/pipelines/content/experiment/ManageExperimentModal.tsx (hunk 2 above)
Comment (DaoDaoNoCode) [reply frontend/src/concepts/pipelines/content/experiment/ManageExperimentModal.tsx:22]: I think it was based on a conversation I had a while ago with @christianvogt because we wanted to refresh the modal state every time we open/close it.

Comment (christianvogt) [reply frontend/src/concepts/pipelines/content/experiment/ManageExperimentModal.tsx:22]: I recall a conversation around using `key` attribute to force re-rendering a component. I don't believe it was related to this scenario.

I also recall a conversation about how to tend to do a lot of "work" while models are closed because of the pattern we've adopted by rendering modal wrapping components such as this which perform the work even though the modal itself is not yet open. Perhaps that's what trigger this? I honestly I don't remember anything specific to this component.

File: frontend/src/concepts/pipelines/content/createRun/contentSections/ExperimentSection.tsx (hunk 1 above)
Comment (andrewballantyne) [reply frontend/src/concepts/pipelines/content/createRun/contentSections/ExperimentSection.tsx:43]: We stopped doing this I think because it snapped the modal in -- so yes, css transitions. It's also not the way we have written our other dozens of modals... so why be different?

File: frontend/src/concepts/pipelines/content/experiment/ManageExperimentModal.tsx (hunk 2 above)
Comment (christianvogt) [reply frontend/src/concepts/pipelines/content/experiment/ManageExperimentModal.tsx:22]: We had a conversation sometime back while peer programming to solve some issues. I don't recall the specifics. There wasn't any direction given about sweeping changes.
Again, changes to this modal i don't recall specifically discussing.

File: frontend/src/concepts/pipelines/content/createRun/contentSections/ExperimentSection.tsx (hunk 1 above)
Comment (christianvogt) [reply frontend/src/concepts/pipelines/content/createRun/contentSections/ExperimentSection.tsx:43]: There's no suggestion in my comment to be different, just asking for specific examples of your statement.

I suggest we continue to do things consistently and log issues to cover any incorrect usage to address across the app.

**Comment by Gkrumbach07** [comment]
code works as expected
/lgtm

**Comment by Gkrumbach07** [comment]
/hold

feature flag needs to block experiments section

File: frontend/src/concepts/pipelines/content/pipelineSelector/PipelineSelectorTableRow.tsx [hunk 5]
```diff
@@ -35,13 +35,7 @@ const PipelineSelectorTableRow: React.FC<PipelineSelectorTableRowProps> = ({ obj
         }
         triggerRef={tooltipRef}
       />
-
-      <Tr
-        ref={tooltipRef}
-        onRowClick={onClick}
-        isClickable
-        data-id="pipeline-selector-table-list-row"
```
Comment (andrewballantyne) [inline frontend/src/concepts/pipelines/content/pipelineSelector/PipelineSelectorTableRow.tsx:43]: why did you drop a data id?

Comment (Gkrumbach07) [reply frontend/src/concepts/pipelines/content/pipelineSelector/PipelineSelectorTableRow.tsx:43]: its not used anywhere in tests. also im pretty sure tests only look for data-testid?

Comment (DaoDaoNoCode) [reply frontend/src/concepts/pipelines/content/pipelineSelector/PipelineSelectorTableRow.tsx:43]: Right, just some small cleanups since I already changed this file.

**Comment by DaoDaoNoCode** [comment]
/unhold
Feature flag added and ready for review.

File: frontend/src/concepts/pipelines/content/pipelineSelector/PipelineSelectorTableRow.tsx (hunk 5 above)
Comment (Gkrumbach07) [reply frontend/src/concepts/pipelines/content/pipelineSelector/PipelineSelectorTableRow.tsx:43]: ok. marking this resolved then

**Comment by Gkrumbach07** [comment]
I tested the feature flag and it blocks in all the create run forms. this looks good to me. Andrew will need to resolve his comments to merge this

/lgtm

File: frontend/src/concepts/pipelines/content/experiment/ManageExperimentModal.tsx (hunk 2 above)
Comment (Gkrumbach07) [reply frontend/src/concepts/pipelines/content/experiment/ManageExperimentModal.tsx]: looks like this was reverted. so should be resolved

File: frontend/src/__mocks__/mockKFExperiment.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (Gkrumbach07) [reply frontend/src/__mocks__/mockKFExperiment.ts]: this was resolved

File: frontend/src/concepts/pipelines/content/createRun/contentSections/ExperimentSection.tsx (hunk 1 above)
Comment (Gkrumbach07) [reply frontend/src/concepts/pipelines/content/createRun/contentSections/ExperimentSection.tsx]: resolved

**Comment by Gkrumbach07** [comment]
adding approval label. 
/approve

**Review by andrewballantyne — APPROVED** [review]
